src/toil/leader.py

```diff
@@ -290,9 +309,14 @@ def processRemovedJob(jobStoreID):
                 else:
                     raise
             if jobWrapper.logJobStoreFileID is not None:
-                logger.warn("The jobWrapper seems to have left a log file, indicating failure: %s", jobStoreID)
                 with jobWrapper.getLogFileHandle( self.jobStore ) as logFileStream:
-                    logStream( logFileStream, jobStoreID, logger.warn )
+                    # more memory efficient than read().striplines() while leaving off the
+                    # trailing \n left when using readlines()
+                    # http://stackoverflow.com/a/15233739
+                    messages = [line.rstrip('\n') for line in logFileStream]
```

Comment:
This should be a generator expression. Did I not mention that in my previous review? Come talk to me if you are unclear about the difference.